layers/+lang/lsp-c-c++/funcs.el

```diff
@@ -0,0 +1,131 @@
+;;; funcs.el --- cquery layer function definitions file for Spacemacs
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
+;;
+;; Author: Cormac Cannon <cormacc-public@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(require 'cl-lib)
+(require 'subr-x)
+
+(defun lsp-c-c++//enable ()
+  (condition-case nil
+    (if (eq lsp-c-c++-backend 'cquery) (lsp-cquery-enable) (lsp-ccls-enable))
+    (user-error nil)))
+
+(defun cquery//enable ()
+  (condition-case nil
+    (lsp-cquery-enable)
+    (user-error nil)))
+
+
+(defun lsp-c-c++//common-config ()
+  (dolist (mode c-c++-modes)
+    (spacemacs/lsp-bind-keys-for-mode mode)
+    (lsp-c-c++//bind-keys-for-mode mode))
+  (lsp-c-c++//customise-lsp-ui-peek)
+  (setq-default flycheck-disabled-checkers '(c/c++-clang c/c++-gcc)))
+
+(defun lsp-c-c++//bind-keys-for-mode (mode)
+  (spacemacs/set-leader-keys-for-major-mode mode
+    ;; format
+    "=c" 'spacemacs/clang-format-region-or-buffer
+    "=f" 'spacemacs/clang-format-function
+    ;; goto
+    "gf" 'find-file-at-point
+    "gF" 'ffap-other-window)
+  (ecase lsp-c-c++-backend
```

Comment:
I would propose to use ecase in other places. It looks much better that if cases.